synfig-studio/src/gui/workarea.cpp

```diff
@@ -1319,11 +1318,10 @@ WorkArea::on_drawing_area_event(GdkEvent *event)
 			break;
 		}
 		case 2:	{ // Attempt to drag and move the window
-			if (Duck::Handle duck = find_duck(mouse_pos, radius))
-				duck->signal_user_click(1)();
-			else
-			if(Bezier::Handle bezier = find_bezier(mouse_pos, radius, &bezier_click_pos))
-				bezier->signal_user_click(1)(bezier_click_pos);
```

Comment:
Why do you delete those lines?

Reply:
this code was responsible for middle mouse clicking ducks, but it doesnt really need to interact so i though it should be removed

Reply:
Oh sorry, I thought you were talking about the middle button effect while dragging ducks...

I think we should keep as it is, but add your code before this one.